### ISSUE TITLE
throw warning for same line variable declarations

### ISSUE THREAD

**Issue by ORESoftware** [eslint/eslint#2599]
I would like my codebase to throw a warning for this type of code:

`var x = 5, y = 7, z = 4; //bad`

also maybe throw a warning for this also 

`var x = 5; var y = 7; var z =4;  // this is bad also (all declars are on same line)`

how can I throw a warning for these types of variable declarations with ESLint?

I cannot find it in the docs..thanks

what I am looking for is to simply enforce this:

```
 var x = 5;   //good 
 var y = 7;   //good
 var z = 4;   //good 
```

**Comment by lo1tuma** [comment]
What about [newline-after-var](http://eslint.org/docs/rules/newline-after-var).

**Comment by ilyavolodin** [comment]
If I'm not mistaken you are looking for [space-infix-ops](http://eslint.org/docs/rules/space-infix-ops)

**Comment by gyandeeps** [comment]
@lo1tuma That rule only ensures a new line after all the `var` declarations. it doesn't take into account space between 2 declarations.
@ilyavolodin I am not sure if that would solve it.

**Comment by ilyavolodin** [comment]
```
/* eslint space-infix-ops: 2 */
var foo =bar;
```
returns
```
1:8 - Infix operators must be spaced. (space-infix-ops)
```

**Comment by xjamundx** [comment]
@ORESoftware can you update your post so that the `=` signs have spacing in the first example :)

**Comment by gyandeeps** [comment]
Thats for infix operator but I think what he wants is to have all the variable declaration on new lines and not on the same line. If I understand it correctly.

**Comment by lo1tuma** [comment]
Related https://github.com/eslint/eslint/issues/1622.

**Comment by gyandeeps** [comment]
yup its the duplicate of 1622.

**Comment by ilyavolodin** [comment]
Ahh.. sorry, misunderstood the question

**Comment by lo1tuma** [comment]
I don't think it is a duplicate, AFAIK #1622 is only about assignments within variable declarations.

**Comment by gyandeeps** [comment]
Again by going with the title of issue 1622, its a duplicate.

**Comment by xjamundx** [comment]
@gyandeeps I think there are 2 asks in this issue.

1. newline after declaration
2. newline after declarator

#1622 and to some extent http://eslint.org/docs/rules/newline-after-var address 2, but 1 is a new ask I think.

**Comment by xjamundx** [comment]
Interesting, it looks like the suggestion in #1622 actually has some version of this, per the rule they hope to copy in JSCS, which would allow this:
```js
var a,
      foo = 7,
      a, b, c,
      bar = 8;
```

Seems like this *is* a dup.

**Comment by nzakas** [comment]
@ORESoftware can you please clarify what you're requesting?

**Comment by ORESoftware** [comment]
guys, I honestly have NO idea idea how to make it more clear :(

`var x=7,y=5;` //BAD!!!

`var x = 7;` //GOOD
`var y = 5;` //GOOD

**Comment by xjamundx** [comment]
@ORESoftware I think you just want the [one-var](http://eslint.org/docs/rules/one-var.html) rule with the `"never"` option? I think that should solve the main use case.

**Comment by gyandeeps** [comment]
Use cases:

1.
```js
var a = 1, b = 2;
```

2.
```js
var a, b;
```

3.
```js
var a = 1, b;
```

4.
```js
var a, b = 1;
```

5.
```js
var a = 1; var b = 1;
```

6.
```js
var a = 1,
      b = 1;
```

7.
```js
var a,
      b;
```

8.
```js
var a;
var b;
```

What options would pass/fail: (here is my take)
1. Fail
2. Fail
3. Fail
4. Fail
5. Fail
6. Pass
7. Pass
8. Pass

Please share your views. Keep adding more options in your comments but make sure you follow the numbering. so that its easy to talk based on the use case number.

**Comment by xjamundx** [comment]
What if 5 was its own rule that was basically like `one-statement-per-line`

**Comment by nzakas** [comment]
@ORESoftware so far, your just saying "good" or "bad", you haven't talked about what rule configuration you've tried, what ESLint is currently doing, etc. I have no idea of you're asking for something new or something we already have. It would be helpful to use the standard issue format:

1. What code do you have?
2. What is the ESLint output?
3. What do you expect the output to be?

**Comment by ORESoftware** [comment]
Thanks I think what I want is covered by the one-var rule
